### PR TITLE
server-2047 | simplified postgres migration

### PR DESCRIPTION
**Pull request by atulsingh0**
:gear: **Issue**

[SERVER-2047](https://circleci.atlassian.net/browse/SERVER-2047)

:white_check_mark: **Fix**

* Auto execution of Postgres commands if using Internal Postgres
* ~~Re-creating the postgres password secret as a precautionary step~~ 
* Cleaned-Up trailing whitespaces
 
:question: **Tests**

- [X] Tested Updating Existing Instance
- [ ] Installed on new instance

### CODE REVIEW

File: kots-exporter/kots-exporter.sh [hunk 1]
```diff
@@ -346,8 +343,32 @@ rm_kots_annot_label_resources(){
...
+    export POSTGRESQL_PASSWORD="$(kubectl get secret --namespace $namespace postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode)"
+    export POSTGRESQL_PVC="$(kubectl get pvc --namespace $namespace -l app.kubernetes.io/instance=circleci-server,role=primary -o jsonpath='{.items[0].metadata.name}')"
+    kubectl delete statefulsets.apps postgresql --namespace $namespace --cascade=orphan
+    kubectl delete secret postgresql --namespace $namespace
+
+    echo ""
+    echo "Postgres PVC Name: $POSTGRESQL_PVC"
+    echo ""
+    echo "Creating & tagging Postgresql Secret"
+    kubectl -n $namespace create secret generic postgresql --from-literal=postgres-password="$POSTGRESQL_PASSWORD"
```
Comment (lloydChris) [inline kots-exporter/kots-exporter.sh:368]: The postgres secret did not have to be created before, why does it need to create it now?

**Review by lloydChris — COMMENTED** [review]
Love the simplification 👍

File: kots-exporter/kots-exporter.sh [hunk 2]
```diff
@@ -346,8 +343,32 @@ rm_kots_annot_label_resources(){
...
+    kubectl -n $namespace delete "$(kubectl -n $namespace get jobs -o name | grep inject)"
+    fi
+}
+
+postgres_migration(){
+    if "$POSTGRES_INTERNAL"; then
+    echo ""
+    echo "############ PREPARING POSTGRES FOR MIGRATION ############"
+
+    echo "Upgrading to server CircleCI Server 4.0 includes upgrading the Postgres chart"
```
Comment (ryanwohara) [inline kots-exporter/kots-exporter.sh:356]: I think server needs to be lowercase here

```suggestion
    echo "Upgrading to server CircleCI server 4.0 includes upgrading the Postgres chart"
```

File: kots-exporter/kots-exporter.sh (hunk 1 above)
Comment (atulsingh0) [reply kots-exporter/kots-exporter.sh:368]: after discussing with @soulchips , It is being created just as a precautionary step and will be overwritten by the helm upgrade command.

File: kots-exporter/kots-exporter.sh [hunk 3]
```diff
@@ -346,8 +343,32 @@ rm_kots_annot_label_resources(){
     echo "Kots removal logs are available - $path/logs/$kotsCleanupLogFile"
 
     echo "Deleting job inject-bottoken-xxxx"
-    kubectl -n $namespace delete "$(kubectl -n $namespace get jobs -o name | grep inject)" \
-    || echo "Manually run the command: kubectl -n $namespace delete $(kubectl -n $namespace get jobs -o name | grep inject)"
+    if [ "$(kubectl -n $namespace get jobs -o name | grep inject)" != "" ]; then
+    kubectl -n $namespace delete "$(kubectl -n $namespace get jobs -o name | grep inject)"
+    fi
```
Comment (jbialy) [inline kots-exporter/kots-exporter.sh:348]:
```suggestion
    if [ "$(kubectl -n $namespace get jobs -o name | grep inject)" != "" ]; then
        kubectl -n $namespace delete "$(kubectl -n $namespace get jobs -o name | grep inject)"
    fi
```

**Review by lloydChris — APPROVED** [review]

**Pull request merged**